react-hot-button-component/src/hot-button.jsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+function hotButton() {
+    let buttonClass = 'hot';
+
+    if(this.state.isClicked >= 18){
+        buttonClass = 'hot6';
+    }
+    if(this.state.isClicked >= 15){
+        buttonClass = 'hot5';
+    }
+    if(this.state.isClicked >= 12){
+        buttonClass = 'hot4';
+    }
+    if(this.state.isClicked >= 9){
+        buttonClass = 'hot3';
+    }
+    if(this.state.isClicked >= 6){
+        buttonClass = 'hot2';
+    }
+    if(this.state.isClicked >= 3){
+        buttonClass = 'hot1'
+        
+    }
+    return <button className={buttonClass} onClick={this.handleClick}>Hot Button</button>

```

Comment:
This is a better approach, however you don't need to make a separate component, you can just put this code in the `render` method of your class component. Also `this.state.isClicked` won't exist in this component.